kingpin/actors/rightscale/server_array.py

```diff
@@ -712,14 +719,40 @@ def _execute(self):
                 task=task,
                 task_name=task_name,
                 sleep=self.option('expected_runtime'),
-                logger=self.log.info
+                logger=self.log.info,
+                meta_data=instance
             ))
 
         self.log.info('Waiting for %s tasks to finish.' % count)
-        success = yield task_waiting
+        status_instances = yield task_waiting
```

Comment:
Ok all of the code below here makes sense, but seems like its enough logic that it should be in its own method for simpler testing and readability.